book/src/project-policies.md

```diff
@@ -1,36 +1,69 @@
-# Kubernetes and CSI Sidecar Compatibility
+# Versioning, Support, and Kubernetes Compatibility
 
-Every version of a sidecar has a minimum, maximum and recommended Kubernetes version
+## Versioning
+
+Each Kubernetes CSI component version is expressed as *x.y.z*, where *x* is the
+major version, *y* is the minor version, and *z* is the patch version,
+following [Semantic Versioning](https://semver.org/).
+
+Patch version releases only contain bug fixes that do not break any backwards
+compatibility.
+
+Minor version releases may contain new functionality that do not
+break any backwards compatibility (except for alpha features).
+
+Major version releases may contain new functionality or fixes that may break
+backwards compatibility with previous major releases. Changes that require a
+major version increase include: removing or changing API, flags, or behavior, new
+RBAC requirements that are not opt-in, new Kubernetes minimum version
+requirements.
+
+A litmus test for not breaking compatibility is to replace the image of a
+component in an existing deployment without changing that deployment in any
+other way.
+
+## Support
+
+The Kubernetes CSI project follows the broader Kubernetes project on support.
+Every minor release branch will be supported with patch releases for at least 1 year,
+starting with the first release of that minor version. In addition, the minor
+release branch will be supported for at least 3 months after the next minor
+version is released, to allow time to integrate with the latest release.
```

Comment:
So if any of these two criteria apply, the branch is still supported? It's one year for release-1.0 since 1.0.0 came out, even if 1.1.0 and release-1.1 get created one month after 1.0.0?

That could lead to a proliferation of branches that all need to be supported even when in practice the newer ones are backward-compatible drop-in replacements for the older ones.

Reply:
The intent here is to give storage vendors a grace period to adapt and qualify the new release while they still have to support the older releases.

In practice, there's two scenarios I see:
* We continue to regularly make releases every quarter. Then this second clause doesn't apply because the 1 year clause is longer and overlaps. In this scenario, an active component that releases 3 times a year could end up needing to support 3 release branches at any one time.
* A particular component is relatively stable and hasn't had any big changes in a year (it's all been patch releases). The number of release branches that are supported could be as low as 1. When time comes for a new minor release, then in this scenario, we would support 2 release branches for 3 months, after which the older one drops out of support.

Reply:
I'm okay with that. I just wanted to call out this corner case and be sure that we interpret the description the same way.